crates/l2/contracts/src/l2/CommonBridgeL2.sol

```diff
@@ -11,16 +12,54 @@ contract CommonBridgeL2 is ICommonBridgeL2 {
         0x000000000000000000000000000000000000FFFE;
     address public constant BURN_ADDRESS =
         0x0000000000000000000000000000000000000000;
+    /// @notice Token address used to represent ETH
+    address public constant ETH_TOKEN =  0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    // Some calls come as a privileged transaction, whose sender is the bridge itself.
+    modifier onlyBridge() {
+        require(msg.sender == address(this), "CommonBridgeL2: caller is not the bridge");
+        _;
+    }
 
     function withdraw(address _receiverOnL1) external payable {
         require(msg.value > 0, "Withdrawal amount must be positive");
 
         (bool success, ) = BURN_ADDRESS.call{value: msg.value}("");
         require(success, "Failed to burn Ether");
 
-        IL1Messenger(L1_MESSENGER).sendMessageToL1(keccak256(abi.encodePacked(
+        IL2ToL1Messenger(L1_MESSENGER).sendMessageToL1(keccak256(abi.encodePacked(
+            ETH_TOKEN,
+            ETH_TOKEN,
             _receiverOnL1,
             msg.value
         )));
     }
+
+    function mintERC20(address tokenL1, address tokenL2, address destination, uint256 amount) external onlyBridge {
+        (bool success, ) = this.call(abi.encodeCall(this._mintERC20, (tokenL1, tokenL2, destination, amount)))
+        if (!success) {
+            _withdrawERC20(tokenL1, tokenL2, destination, amount);
+        }
+    }
+
+    function _mintERC20(address tokenL1, address tokenL2, address destination, uint256 amount) external onlyBridge {
```

Comment:
This function is `external`. We should rename it to something like `tryMintERC20`. Alternatively, we could make it `private` and make it return a boolean instead of reverting.